src/__tests__/components/Table.test.js

```diff
@@ -234,4 +234,52 @@ describe('Table', () => {
         expect(lastName.sortable).toBe(false);
         expect(email.sortable).toBe(true);
     });
+
+    describe('when server side sorting is used', () => {
+        let updateDataSpy, pagination = {}, sort = {};
+
+        beforeEach(() => {
+            updateDataSpy = jest.fn();
+            props = {
+                ...props,
+                updateData: updateDataSpy
+            };
+            wrapper = mount(<Table {...props} />)
+            instance = wrapper.instance();
+        });
+
+        describe('sortRows()', () => {
+            it('should call updateData with sort column and reversed direction', () => {
+                pagination = {
+                    currentPage: 1,
+                    totalPages: 2,
+                }
+                sort = {
+                    column: 'firstName',
+                    direction: 'ascending'
+                }
+                wrapper.setState({ pagination, sort });
+                instance.sortRows({ column: 'firstName' });
+
+                expect(updateDataSpy).toHaveBeenCalledTimes(1);
+                expect(updateDataSpy).toHaveBeenCalledWith({page: pagination.currentPage, sort: {...sort, direction: 'descending'}});
+            });
+
+            it('should call updateData with updated sort column and the default ascending direction', () => {
```

Comment:
It's not clear to me why the first test results in descending data, and the second in ascending data.
As near as I can tell, the set-up for both tests looks the same (i.e. the sort object both have the same initial `direction`).
Think that needs some explanation.

Reply:
I will change the wording a little. The first test basically tests that if the sort direction is reversed when the same column is requested to be sorted again. And the second one tests that when a new column is selected to be sorted, the new column is set the sort column and direction is set to ascending by default.